### PR TITLE
Refactor osu!mania skinning to allow skins to specify column accent colours

### PR DESCRIPTION
**Pull request by peppy**
I've tried for a few hours to split this out into smaller pieces, but it's a real pain to do so. In addition, I'm not sure whether there is going to be any issue taken with the refactors I've made (or if I've overlooked something important) so I want to get this out for review before spending more time attempting to make it digestible.

Of note:

- Argon column colours are just RNG for now. Still waiting for colour spec from design.
- I'm still not sure on the future of `StageDefinition`, but it seems like it may grow in the future.
- I removed `ColumnType` because argon is supposed to have a hue range, not just odd/even column colouring. The logic which used this has been moved into the `Triangles` and `Legacy` transformers
- I can see a day where we can remove the `LegacyManiaSkinConfigurationLookup` altogether (replace with some code inside the transformers). But this is definitely for another day.

Also:

- [x] Depends on #20571 (conflict avoidance)

### CODE REVIEW

**Comment by smoogipoo** [comment]
Needs conflict resolution

**Comment by peppy** [comment]
Rather than conflict resolve I've just rebased and force pushed on top of latest `master` (results in clean application).

**Comment by peppy** [comment]
Also to note, in follow up work I've found that changing skins during gameplay doesn't always update the accent colour of all hitobjects correctly. @smoogipoo if you can see why during review then please let me know (I may be using the incorrect application to `Objects` in the non-bindable flow).

The more correct path forward is probably binding to the DHOs..

**Comment by peppy** [comment]
One more thing I found is that some beatmaps (with beatmap skin enabled) will cause stage colours to get overwritten. I'm looking into this.

File: osu.Game.Rulesets.Mania/UI/Components/DefaultKeyArea.cs [hunk 1]
```diff
@@ -75,15 +77,19 @@ private void load(IScrollingInfo scrollingInfo)
...
+            accentColour = column.AccentColour.GetBoundCopy();
+            accentColour.BindValueChanged(colour =>
+            {
+                keyIcon.EdgeEffect = new EdgeEffectParameters
+                {
+                    Type = EdgeEffectType.Glow,
+                    Radius = 5,
+                    Colour = colour.NewValue.Opacity(0.5f),
+                };
+            });
```
Comment (smoogipoo) [inline osu.Game.Rulesets.Mania/UI/Components/DefaultKeyArea.cs:92]:
```suggestion
            }, true);
```

**Review by smoogipoo — CHANGES_REQUESTED** [review]

**Review by smoogipoo — APPROVED** [review]

**Pull request merged**